spanner/session.go

```diff
@@ -570,12 +595,20 @@ type sessionPool struct {
 	// idleList caches idle session IDs. Session IDs in this list can be
 	// allocated for use.
 	idleList list.List
+	// multiplexedSessions contains the multiplexed sessions
+	multiplexedSession *session
 	// mayGetSession is for broadcasting that session retrival/creation may
 	// proceed.
 	mayGetSession chan struct{}
+	// mayGetSession is for broadcasting that multiplexed session retrival/creation may
+	// proceed.
+	mayGetMultiplexedSession chan struct{}
 	// sessionCreationError is the last error that occurred during session
 	// creation and is propagated to any waiters waiting for a session.
 	sessionCreationError error
+	// multiplexedSessionCreationError is the last error that occurred during multiplexed session
+	// creation and is propagated to any waiters waiting for a session.
+	multiplexedSessionCreationError error
```

Comment:
(Adding a comment for now, maybe it turns out to be void after reading the rest of the code)

We should only propagate this error to any waiters for the first creation of a multiplexed session. If refreshing a multiplexed session fails, then we should not propagate that error to the application, as the multiplexed session is likely to be usable for many days still.